doc/task-hierarchy.md

```diff
@@ -0,0 +1,164 @@
+Task Type
+- Contextual Question Generation
+   - Generate questions based on given context e.g. a task to create a question based on a paragraph.
+- Answer Generation: Reading Comprehension
+- Answer Generation: Contextual Question Answering
+- Answer Generation: Open book/Domain-specific Question Answering
+- Answer Generation: Commonsense Question Answering
+- Answer Generation: Fill in the Blank
+- Answer Generation: Multiple Choice Question Answering
+- Textual Entailment
+- Incorrect Answer Generation
+- Sentence Generation
+- Paragraph Generation
+- Structured Text Generation
+   - Generate structured text in the output e.g. a task that converts questions in natural language to SQL queries
+- Topic Generation
+- Explanation Generation
+- Review Generation
+- Order Generation
+  - Given a set of elements, find their order (e.g. monotonically increasing/decreasing numbers, increasing/decreasing size in case of objects)
+- Title Generation
+- Weblink Generation
+- Text Modification
+- Classification
+- Verification 
+   - Verify if two inputs match with each other e.g. a task to verify if a sentence and its sentiment given in the input match
+- Translation
+- Hypernym Discovery
+- Parts-of-speech
+- Coreference: event coreference
+- Intent Detection
+- Claim Verification
+- Paraphrasing
+- Relation Prediction
+- Analogical Reasoning
+- Semantic Parsing
+- Stance Detection
+- Word Sense Disambiguation
+- Mathematics: Arithmetic
+- Mathematics: Algebra
+- Mathematics: Geometry
+- Question Decomposition
+- Counting
+   - Count an attribute of input e.g. a task to count number of vowels in a given word
+- Entity Detection
+- Role Labelling
+- Combinatorics
+- Grammar Error Correction
+- Summarization
+- Sentiment Analysis
+- Hate Speech Detection
+- Fake News Detection
+- Ethical Judgement
+- Style Transfer
+- Text Comparison
+- Question Composition
+   - Compose questions by concating questions in the input
+- Column Matching
+   - Given two sets in the input, generate a mapping between them e.g. given a set of countries and their capitals in the input, generate an output that maps countries to capitals.
+- Handling Commands
+- Author Identification
+- Sorting
+- Language Identification
+- Text Span Selection
+- Hallucination
+   - Given a context, generate imaginary content e.g. given a sentence, generate a story/poem.
+- Document Classification
+- Relevancy Estimation
+- Named Entity Recognition
+- Dialogue Understanding
+- Abuse Detection
+- Text Simplification
+- Tabular Text Operation
```

Comment:
"Tabular Text Operation" and "Column Matching" seem to be relevant. Should they have a common ancestor?